src/sage/tensor/modules/comp.py

```
@@ -253,6 +253,58 @@
 from sage.parallel.parallelism import Parallelism
 from operator import itemgetter
 
+def get_components_class(sym=None, implementation=None) -> type:
+    r"""
+    Return a component class according to the input.
+
+    INPUT:
+
+    - ``sym`` -- (``None`` or string) symmetry of component class.
+
+    - ``implementation`` -- (``None`` or string) a possible implementation.
+    """
+    if implementation is None:
+        if sym is None:
+            return Components
+        elif sym == "sym":
+            return CompWithSym
+        elif sym == "fullysym":
+            return CompFullySym
+        elif sym == "fullyantisym":
+            return CompFullyAntiSym
+        elif sym == "kroneckerdelta":
+            return KroneckerDelta
+
+    if implementation == "numpy":
+        try:
+            from . import comp_numpy
+        except ImportError:
+            pass
+        if sym is None:
+            return comp_numpy.ComponentNumpy
+        elif sym == "sym":
+            return comp_numpy.CompNumpyWithSym
+        elif sym == "fullysym":
+            return comp_numpy.CompNumpyFullySym
+        elif sym == "fullyantisym":
+            return comp_numpy.CompNumpyFullyAntiSym
+        elif sym == "kroneckerdelta":
+            return comp_numpy.KroneckerDeltaNumpy
```

probably should raise an error if `sym` is something else